src/index.jsx

```diff
@@ -11,45 +11,49 @@ class CalendarHeatmap extends React.Component {
   constructor(props) {
     super(props);
 
+    this.latestProps = props;
+
     this.state = {
-      valueCache: this.getValueCache(props.values),
+      valueCache: this.getValueCache(this.latestProps.values),
     };
   }
 
   componentWillReceiveProps(nextProps) {
+    this.latestProps = nextProps;
+
     this.setState({
-      valueCache: this.getValueCache(nextProps.values),
+      valueCache: this.getValueCache(this.latestProps.values),
     });
   }
 
   getDateDifferenceInDays() {
-    const { startDate, endDate, numDays } = this.props;
+    const { startDate, endDate, numDays } = this.latestProps;
     if (numDays) {
       // eslint-disable-next-line no-console
       console.warn('numDays is the deprecated prop, so it will be removed in the next release. Consider using of startDate prop instead.');
       return numDays;
     }
     const timeDiff = endDate - startDate;
-    return Math.ceil(timeDiff / (1000 * 3600 * 24));
+    return Math.ceil(timeDiff / MILLISECONDS_IN_ONE_DAY);
```

Comment:
I updated the `getDateDifferenceInDays` method to use the `MILLISECONDS_IN_ONE_DAY` constant instead of calculating the same value itself. This change is unrelated to the rest of this PR.